### DISCUSSION TITLE
How to get the struct identifier in a definition?

### DISCUSSION THREAD

**Discussion by yann-soubeyrand** [cue-lang/cue#1859] [Q&A]
Hello,

In a definition, is it possible to get the identifier of the struct which is unifying to this definition? Given this config
```
#Def: self = {
  args: [...string]
  command: {
    name: <what to put here?>
    args: self.args
  }
}

myCommand: #Def & {
  args: [
    "arg1",
    "arg2",
  ]
}
```

is it possible to set myCommand.command.name to "myCommand"?

**Comment by slewiskelly** [comment]
You can use [templates](https://cuelang.org/docs/tutorials/tour/types/templates/):

```cue
#Def: self={
	args: [...string]
	command: {
		name: string
		args: self.args
	}
}

[Command=string]: #Def & {command: name: Command}

myCommand: #Def & {
	args: [
		"arg1",
		"arg2",
	]
}
```

**Comment by yann-soubeyrand** [comment]
Thanks for your answer @slewiskelly. I was asking because I had two issues with your proposed solution:
- You have to add the pattern at the place where you instantiate the definition and cannot do it at the place where the definition is defined.
- The other issue is: https://github.com/cue-lang/cue/discussions/1856.
